### PR TITLE
Fix some broken and redirected links

### PR DESCRIPTION
**Pull request by namurphy**
This PR makes a few quick corrections to web links that were either broken or were redirects.  One of the changes is for an old changelog entry that had linked to the issue it closed rather than the pull request number.  I found these out doing `make linkcheck` in `docs/` which was pretty handy.  (I made a reST update here and there while I was the area too.)

### CODE REVIEW

File: plasmapy/formulary/collisions.py [hunk 1]
```diff
@@ -1018,6 +1020,7 @@ def Coulomb_cross_section(impact_param: u.m) -> u.m ** 2:
     References
     ----------
     .. [1] https://en.wikipedia.org/wiki/Cross_section_(physics)#Collision_among_gas_particles
+
```
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1023]:
```suggestion
```

File: plasmapy/formulary/collisions.py [hunk 2]
```diff
@@ -1104,8 +1107,9 @@ def fundamental_electron_collision_freq(
        revised." Naval Research Lab. Report NRL/PU/6790-16-614 (2016).
        https://www.nrl.navy.mil/ppd/content/nrl-plasma-formulary
 
-    .. [3] J.D. Callen, Fundamentals of Plasma Physics draft material,
-       Chapter 2, http://homepages.cae.wisc.edu/~callen/chap2.pdf
+    .. [3] `Draft Material for "Fundamentals of Plasma Physics" Book
+       <https://docs.google.com/document/d/e/2PACX-1vQmvQ_b8p0P2cYsWGMQYVd92OBLX9Sm6XGiCMRBidoVSoJffj2MBvWiwpix46mqlq_HQvHD5ofpfrNF/pub>`__,
+       by James D. Callen
```
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1112]: This is a pretty fragile link, so I'm tempted to replace this a more persistent publication later on.  (It also gets repeated a bunch of times so I had to copy & paste a bunch, which gives me more motivation for a unified set of references as in #1245.)

**Comment by namurphy** [comment]
This should be a pretty quick review.  If it looks good to you, please feel free to merge.

File: plasmapy/formulary/collisions.py [hunk 3]
```diff
@@ -1017,7 +1019,8 @@ def Coulomb_cross_section(impact_param: u.m) -> u.m ** 2:
 
     References
     ----------
-    .. [1] https://en.wikipedia.org/wiki/Cross_section_(physics)#Collision_among_gas_particles
+    .. [1] `Cross Section: Collision among gas particles
+       <https://en.wikipedia.org/wiki/Cross_section_(physics)#Collision_among_gas_particles>`__
```
Comment (StanczakDominik) [inline plasmapy/formulary/collisions.py:1023]: Speaking of fragile links, this one should be a permalink:

```suggestion
    .. [1] `Cross Section: Collision among gas particles
       <https://en.wikipedia.org/w/index.php?title=Cross_section_(physics)&oldid=1037954726#Collision_among_gas_particles>`__
```

**Review by StanczakDominik — APPROVED** [review]
Looks good, two minor additions!

**Pull request merged**